### PR TITLE
Updated .NET SDK from latest swagger for RecoveryServices.Backup

### PR DESCRIPTION
**Pull request by DheerendraRathor**
## Description

- Upgraded version to 2.0.0-preview
- Removed redundant files
- Added tests and record files for List Operations API
- Updated ResourceManager to 1.6.0-preview

Swagger spec PR: https://github.com/Azure/azure-rest-api-specs/pull/1267 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/Models/Page2.cs [hunk 1]
```diff
@@ -28,10 +28,10 @@ public class Page2<T> : IPage<T>
          /// <summary>
          /// Gets the link to the next page.
          /// </summary>
-         [JsonProperty("")]
+         [JsonProperty("NextLink")]
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/Models/Page2.cs]: @DheerendraRathor why is it not camelCase?

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/Models/Page2.cs [hunk 2]
```diff
@@ -28,10 +28,10 @@ public class Page2<T> : IPage<T>
...
          public string NextPageLink { get; private set; }
 
-         [JsonProperty("value")]
+         [JsonProperty("Value")]
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/Models/Page2.cs]: @DheerendraRathor any reason this is not camelCase?

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Microsoft.Azure.Management.RecoveryServices.Backup.csproj [hunk 3]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.Management.RecoveryServices.Backup</PackageId>
     <Description>Provides developers with libraries for the updated recovery services backup feature under Azure Resource manager.</Description>
-    <VersionPrefix>1.3.0-preview</VersionPrefix>
+    <VersionPrefix>1.4.0-preview</VersionPrefix>
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Microsoft.Azure.Management.RecoveryServices.Backup.csproj]: @DheerendraRathor you have a breaking change, that requires major version change and not minor version.

Please also use latest resource manager nuget package for recording your tests that you have in this PR.
You are still using 1.1.0-preview Resource Manager and you need start using >=1.6.0

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -8,7 +8,7 @@
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Recovery Services Backup Resources.")]
 
 [assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.3.0.0")]
+[assembly: AssemblyFileVersion("1.4.0.0")]
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Properties/AssemblyInfo.cs]: @DheerendraRathor major version change in FileVersion as well as AssemblyVersion

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/Models/Page2.cs (hunk 1 above)
Comment (DheerendraRathor) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/Models/Page2.cs]: This is auto generated from swagger spec and which in turn reflects field name directly from service data contracts.

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/Models/Page2.cs (hunk 2 above)
Comment (DheerendraRathor) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/Models/Page2.cs]: Same as for "NextLink"

**Comment by DheerendraRathor** [comment]
@shahabhijeet Currently project is referring `Microsoft.Azure.ResourceManager 1.1.0-preview` which is read from `AzSdk.test.reference.props`. On nuget this package is unavailable and name of package is [`Microsoft.Azure.Management.ResourceManager`](https://www.nuget.org/packages/Microsoft.Azure.Management.ResourceManager/1.6.0-preview). Should I still refer to new package? I can't remove old 1.1.0 as it will require modification of `AzSdk.test.reference.props` and that will change references in all test projects.

**Comment by shahabhijeet** [comment]
@DheerendraRathor you add the new package in your test project and not in reference.props file.
Version 1.1.0 and 1.6.0 can co-exists.
The idea is to make sure you are creating and using the latest resource manager package and using it in your tests.

**Comment by DheerendraRathor** [comment]
@shahabhijeet Can you please look into it? It has been pending for a while now.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**